lightning/src/ln/channelmanager.rs

```diff
@@ -968,8 +970,8 @@ impl<ChanSigner: ChannelKeys, M: Deref> ChannelManager<ChanSigner, M> where M::T
 				})
 			} else {
 				let mut new_packet_data = [0; 20*65];
-				chacha.process(&msg.onion_routing_packet.hop_data[65..], &mut new_packet_data[0..19*65]);
-				chacha.process(&SIXTY_FIVE_ZEROS[..], &mut new_packet_data[19*65..]);
+				let read_pos = chacha_stream.read(&mut new_packet_data).unwrap();
+				chacha_stream.chacha.process_inline(&mut new_packet_data[read_pos..]);
```

Comment:
Is it possible to encapsulate this into a single call to ChaChaReader?

Reply:
Hmmm...they're two pretty separate things...one is emptying the buffer of bytes we got from our peer, the other is filling any remaining bytes with 0-bytes encrypted. We could theoretically have a stream that reads from a given buffer and then returns 0s forever, but that just feels like magic that would make the code overly confusing. I did add a comment here noting wtf is going on cause I found it confusing as-is.